app/models/miq_queue_worker_base/runner.rb

```diff
@@ -74,7 +78,7 @@ def get_message_via_sql
     loop do
       msg = MiqQueue.get(
         :queue_name => @worker.queue_name,
-        :role       => @active_roles,
+        :role       => worker_roles,
```

Comment:
Do you have examples rails logs/queries showing a specific worker querying for work for other roles?

Reply:
Of course, here is a query by the event_handler (queue_name => 'ems') with all of the active roles
```
D, [2019-11-11T10:12:27.525848 #13271:2b07977ee5cc] DEBUG -- :   ^[[1m^[[36mMiqQueue Load (0.4ms)^[[0m  ^[[1m^[[34mSELECT  id, lock_version, priority, role FROM "miq_queue" WHERE (    state = 'ready'
    AND (zone IS NULL OR zone = 'default')
    AND (role IS NULL OR role IN ('automate','database_operations','database_owner','ems_inventory','ems_operations','event','remote_console','reporting','scheduler','smartstate','user_interface', 'web_services'))
    AND (server_guid IS NULL OR server_guid = '271497c5-8ade-4451-aa4c-e307907d35cd')
    AND (deliver_on IS NULL OR deliver_on <= '2019-11-11 15:12:27.524998')
    AND (priority <= 200)
    AND queue_name = 'ems'
```